diff --git a/widget/moz.build b/widget/moz.build
index a845e87076d1..920723a30efa 100644
--- a/widget/moz.build
+++ b/widget/moz.build
@@ -196,6 +196,7 @@ EXPORTS.mozilla += [
     "MiscEvents.h",
     "MouseEvents.h",
     "NativeKeyBindingsType.h",
+    "NativeTextInput.h",
     "SwipeTracker.h",
     "TextEventDispatcher.h",
     "TextEventDispatcherListener.h",
@@ -394,7 +395,7 @@ elif toolkit == "android":
         "android/PlatformWidgetTypes.ipdlh",
         "headless/HeadlessWidgetTypes.ipdlh",
     ]
-elif toolkit == "cocoa":
+elif toolkit in ("cocoa", "uikit"):
     IPDL_SOURCES = [
         "cocoa/PCompositorWidget.ipdl",
         "cocoa/PlatformWidgetTypes.ipdlh",
