diff --git a/editor/libeditor/AutoClonedRangeArray.cpp b/editor/libeditor/AutoClonedRangeArray.cpp
index 67426d7b08e6..077c28650bfb 100644
--- a/editor/libeditor/AutoClonedRangeArray.cpp
+++ b/editor/libeditor/AutoClonedRangeArray.cpp
@@ -1609,6 +1609,20 @@ AutoClonedSelectionRangeArray::ExtendAnchorFocusRangeFor(
           "nsFrameSelection::CreateRangeExtendedToNextHardLineBreak() failed");
       directionAndAmountResult = nsIEditor::eNext;
       break;
+    case nsIEditor::eToEndOfParagraph:
+      result =
+          nsFrameSelection::CreateRangeExtendedToNextParagraphBoundary<nsRange>(
+              *presShell, limitersAndCaretData, anchorFocusRange,
+              rangeDirection);
+      if (NS_WARN_IF(aEditorBase.Destroyed())) {
+        return Err(NS_ERROR_EDITOR_DESTROYED);
+      }
+      NS_WARNING_ASSERTION(
+          result.isOk(),
+          "nsFrameSelection::CreateRangeExtendedToNextParagraphBoundary() "
+          "failed");
+      directionAndAmountResult = nsIEditor::eNext;
+      break;
     default:
       return aDirectionAndAmount;
   }
